unstructured_platform_plugins/etl_uvicorn/api_generator.py

```diff
@@ -118,8 +136,14 @@ async def wrap_fn(func: Callable, kwargs: Optional[dict[str, Any]] = None) -> In
         else:
             logger.warning("usage data not an expected parameter, omitting")
         try:
-            output = await invoke_func(func=func, kwargs=request_dict)
-            return InvokeResponse(usage=usage, status_code=status.HTTP_200_OK, output=output)
+
+            async def _stream_response():
+                async for output in invoke_func(func=func, kwargs=request_dict):
+                    yield InvokeResponse(
+                        usage=usage, status_code=status.HTTP_200_OK, output=output
+                    ).model_dump_json() + "\n"
+
+            return StreamingResponse(_stream_response(), media_type="application/x-ndjson")
```

Comment:
Might want to dynamically change the response type of this method to `StreamingResponse` if we know that's what's going to be returned. Also if this block fails, if the response should also be a `StreamingResponse` of just one `InvokeResponse` in the except block?

Reply:
Yes, I think you are right in that. It might be best to only do this if the func is async generator. I'll tweak it.